cadc-vos-server/src/main/java/ca/nrc/cadc/vos/server/transfers/TransferUtil.java

```diff
@@ -229,6 +238,73 @@ else if ("https".equalsIgnoreCase(scheme))
         return protocolList;
     }
 
+    /**
+     * Generate a Protocol list with endpoints set to /vault/pkg/<jobid>. One entry for every protocol in
+     * the transfer provided. Use the jobID from the job provided.
+     * @param transfer
+     * @param job
+     * @return
+     * @throws MalformedURLException
+     * @throws MalformedURLException
+     */
+    public static List<Protocol> getPackageEndpoints(final Transfer transfer, final Job job)
+        throws IOException, IllegalArgumentException, ResourceNotFoundException {
+
+        // package view is redirected to /vault/pkg/<jobid>
+        // making an endpoint for each protocol provided
+
+        List<Protocol> protocolList = transfer.getProtocols();
+        List<Protocol> augmentedList = new ArrayList<>();
+        // Sort the protocols
+        Collections.sort(protocolList, new ProtocolComparator());
```

Comment:
I think this is fine and will work, but I feel like the sort should happen on the augmentedList at the end instead...

Reply:
moved.